src/Atmos/Model/AtmosModel.jl

```diff
@@ -425,10 +425,19 @@ equations.
     flux.ρ = Σfluxes(eq_tends(Mass(), m, tend), args...)
     flux.ρu = Σfluxes(eq_tends(Momentum(), m, tend), args...) .* ρu_pad
     flux.ρe = Σfluxes(eq_tends(Energy(), m, tend), args...)
+    if (m.moisture isa DryModel)
+        flux.moisture.ρq_tot =
+            Σfluxes(eq_tends(TotalMoisture(), m, tend), args...)
+    end
+    if (m.moisture isa NonEquilMoist)
+        flux.moisture.ρq_liq =
+            Σfluxes(eq_tends(LiquidMoisture(), m, tend), args...)
+        flux.moisture.ρq_ice =
+            Σfluxes(eq_tends(IceMoisture(), m, tend), args...)
+    end
```

Comment:
I'd prefer we continue to use `m.moisture` as a dispatch type rather than test it like this.

Reply:
How so? Do you mean leave
```julia
    flux_moisture!(m.moisture, m, flux, state, aux, t)
```
And then call the sum inside `flux_moisture!`? If so, I'm fine with that.

Reply:
I think that's cleaner, yeah.

Speaking more generally, perhaps having a more concrete relationship between model and components, and thereby tendencies, would help make all this neater. Maybe `MoistureModel` etc. should all be subtypes of an `AbstractAtmosComponent` or something like that? Not too sure how best to do that and for a separate PR anyway.